### PR TITLE
Delegate loading .editorconfig properties to ec4j library.

### LINKED ISSUE #742: Cannot turn off one rule for one package
I want to disable one of my custom rules for one package.
## Expected Behavior
Files in package api can ignore rule class-must-be-internal.

## Observed Behavior
Linter apply rule class-must-be-internal for all files.

## Steps to Reproduce
My `.editconfig`

```
[*.{kt, kts}]
disabled_rules = import-ordering, no-wildcard-imports, comment-spacing

[**/api/*.{kt, kts}]
disabled_rules = class-must-be-internal
```

## Your Environment
* Version of ktlint used: 
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): org.jmailen.gradle:kotlinter-gradle:2.3.2
* Version of Gradle used (if applicable): 5.6.4
* Operating System and version: MacOS 10.15.4 (19E287)
* Link to your project (if it's a public repository):

### PR DESCRIPTION
**Pull request by Tapchicoma**
Remove custom logic how to load and cache editorconfig properties for specific file and use ec4j library for it.

For now implementation just uses raw properties values. In the future, we could use custom Properties to also do validation of .editorconfig.

Fixes #742

### CODE REVIEW

File: ktlint-core/src/main/kotlin/com/pinterest/ktlint/core/KtLint.kt [hunk 1]
```diff
@@ -177,58 +179,6 @@ object KtLint {
...
-                        printEditorConfigChain(it) {
-                            editorConfigSet.put(it.path, true) != true
-                        }
-                    }
-                    ?: emptyMap<String, String>()
-                ) + ("file_path" to fileName)
-        }
-    }
-
-    private fun printEditorConfigChain(ec: EditorConfigInternal, predicate: (EditorConfigInternal) -> Boolean = { true }) {
```
Comment (Tapchicoma) [inline ktlint-core/src/main/kotlin/com/pinterest/ktlint/core/KtLint.kt:223]: ec4j does not provide any logging capabilities, so I just dropped this debug message.

**Comment by Tapchicoma** [comment]
FYI: `{kt,  kts}` is not supported by official `.editorconfig` spec - see https://github.com/editorconfig/editorconfig/issues/148

File: ktlint-core/src/main/kotlin/com/pinterest/ktlint/core/internal/ThreadSafeEditorConfigCache.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import kotlin.concurrent.write
+import org.ec4j.core.Cache
+import org.ec4j.core.EditorConfigLoader
+import org.ec4j.core.Resource
+import org.ec4j.core.model.EditorConfig
+
+/**
+ * In-memory [Cache] implementation that could be safely accessed from any [Thread].
+ */
+internal class ThreadSafeEditorConfigCache : Cache {
```
Comment (romtsn) [inline ktlint-core/src/main/kotlin/com/pinterest/ktlint/core/internal/ThreadSafeEditorConfigCache.kt:14]: not sure if it's worth to test this class?

Comment (Tapchicoma) [reply ktlint-core/src/main/kotlin/com/pinterest/ktlint/core/internal/ThreadSafeEditorConfigCache.kt:14]: It is pretty straightforward, but I will add some tests :+1:

File: ktlint-core/src/main/kotlin/com/pinterest/ktlint/core/internal/EditorConfigLoader.kt [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        }
+
+        val normalizedFilePath = when {
+            filePath.endsWith(STDIN_FILE) -> Paths
+                .get(".")
+                .toAbsolutePath()
+                .resolve("stdin${SUPPORTED_FILES.first()}")
+            alternativeEditorConfig != null -> alternativeEditorConfig
+                .toAbsolutePath()
+                .resolve("fake${SUPPORTED_FILES.first()}")
```
[the remark this answers is not in the document]
Comment (Tapchicoma) [reply ktlint-core/src/main/kotlin/com/pinterest/ktlint/core/internal/EditorConfigLoader.kt]: Actually I went to use original file name here - for the cases when `.kt` and `.kts` sections may differ.

File: ktlint-core/src/main/kotlin/com/pinterest/ktlint/core/internal/ThreadSafeEditorConfigCache.kt (hunk 2 above)
Comment (Tapchicoma) [reply ktlint-core/src/main/kotlin/com/pinterest/ktlint/core/internal/ThreadSafeEditorConfigCache.kt:14]: After some thoughts I've decided to not write tests for this class. Class itself is pretty straightforward, though writing non-flaky multithreading tests are hard and not worth, in this particular case, an effort.

**Pull request merged**